### ISSUE TITLE
More advanced ways of collecting metrics in tf.keras while training models

### ISSUE THREAD

**Issue by nbro** [tensorflow/tensorflow#36372]
**System information**
- TensorFlow version (you are using): 2.1
- Are you willing to contribute it (Yes/No): Maybe

**Describe the feature and the current behavior/state.**

Currently, in `tf.keras`,  we can access the metrics by accessing the attribute `metrics` (a list) and `metrics_names` (list) of the model. Furthermore, when developing a callback, e.g. in the definition `def on_epoch_end(epoch, logs)`, `logs` is a dictionary whose keys are the names and whose values the corresponding values of the metrics, including the loss. 

If you have a neural network with multiple outputs, all the metrics for each of these separate outputs are elements of `metrics` (or `metric_names`) and the dictionary `logs`. AFAIK, there's no direct way of getting e.g. the metrics for an output with name `"outpu1"`. This feature would be useful in the case we want to plot all (or a subset of) the metrics associated with one output. The model object could provide a method or attribute, `get_metrics_of_output(output_name)`, where the metrics are returned as an object (e.g. a dictionary) where the metrics associated with one output are collected in one separate list (or dictionary, etc) than the metrics associated with another output, so that `get_metrics_of_output(output_name)` returns the metrics of the output layer with name `output_name`. Furthermore, there should also be a way of differentiating between certain metrics associated with the **same** output. This could be determined by the way we define and pass the metrics to the `metrics` parameter of the `compile` method (e.g. as a dictionary). 

In [this Stack Overflow question](https://stackoverflow.com/q/59996177/3924118), I am describing my current issue where this feature could be useful.

**Will this change the current api? How?**

Probably. You could just introduce a new method that returns a more advanced view of the metrics. `model.metrics` could still return just the flattened version of the metrics (as it is currently the case). However, `logs` (e.g. in `def on_epoch_end(epoch, logs)`) probably should change to return the more advanced view of the metrics. Or maybe there should be a third parameter that returns this advanced view of the metrics, e.g. `def on_epoch_end(epoch, logs, metrics)`.

**Who will benefit with this feature?**

Everyone.

**Any Other info.**

**Comment by rmothukuru** [comment]
@nbro,
Can you please take a look at [Tensorflow Model Analysis](https://www.tensorflow.org/tfx/model_analysis/get_started), which provides a facility to play around with **`Metrics`** of a **`Machine/Deep Learning Model`**, and see if it helps? Thanks!
